R/descriptives.R

```diff
@@ -0,0 +1,161 @@
+# distribution_mode ----------------------------------
+
+#' Compute mode for a statistical distribution
+#'
+#' @param x An atomic vector, a list, or a data frame.
+#'
+#' @return
+#'
+#' The value that appears most frequently in the provided data.
+#' The returned data structure will be the same as the entered one.
+#'
+#' @seealso For continuous variables, the
+#'   **Highest Maximum a Posteriori probability estimate (MAP)** may be
+#'   a more useful way to estimate the most commonly-observed value
+#'   than the mode. See [bayestestR::map_estimate()].
+#'
+#' @examples
+#'
+#' distribution_mode(c(1, 2, 3, 3, 4, 5))
+#' distribution_mode(c(1.5, 2.3, 3.7, 3.7, 4.0, 5))
+#'
+#' @export
+distribution_mode <- function(x) {
+  # TODO: Add support for weights, trim, binned (method)
+  uniqv <- unique(x)
+  tab <- tabulate(match(x, uniqv))
+  idx <- which.max(tab)
+  uniqv[idx]
+}
+
+#' Compute the coefficient of variation
+#'
+#' Compute the coefficient of variation (CV, ratio of the standard deviation to
+#' the mean, \eqn{\sigma/\mu}) for a set of numeric values.
+#'
+#' @return The computed coefficient of variation for `x`.
+#' @export
+#'
+#' @examples
+#' coef_var(1:10)
+#' coef_var(c(1:10, 100), method = "median_mad")
+#' coef_var(c(1:10, 100), method = "qcd")
+#' coef_var(mu = 10, sigma = 20)
+#' coef_var(mu = 10, sigma = 20, method = "unbiased", n = 30)
+#' cv(1:10)
+coef_var <- function(x, ...) {
+  UseMethod("coef_var")
+}
+
+#' @name cv
+#' @rdname coef_var
+#' @export
+cv <- coef_var
```

Comment:
I am wondering if we should support this alias. It is quite non-descriptive, and can be easily confused with cross-validation.

Reply:
Yeah I'm torn. "CV" is by far the most common way to write the stat name (analogous to "SD"), but yeah it's not that descriptive and is easily confused with cross validation. 

I guess I lean toward not including. Thoughts @DominiqueMakowski @mattansb @strengejacke ?

Reply:
@DominiqueMakowski @mattansb @strengejacke @etiennebacher WDYT?

Reply:
`coef_var` is explicit and short enough to me, and I agree that `cv` can be confused with cross validation

Reply:
agree, we should remove this alias.